### DISCUSSION TITLE
Consumer unprocessed messages count is higher than stream messages count.

### DISCUSSION THREAD

**Discussion by Inkathu** [nats-io/nats.net.v1#905] [Q&A]
Hi!
How could it be, that consumer unprocessed messages count is higher than stream messages?
I have the stream:
`Information for Stream OrchestrationStream_08fa7d3dcd644745bace255d2c76fbc2 created 2024-06-20 12:58:59

             Subjects: OrchestrationStream_08fa7d3dcd644745bace255d2c76fbc2.>
             Replicas: 1
              Storage: File

Options:

            Retention: Limits
     Acknowledgements: true
       Discard Policy: New
     Duplicate Window: 8760h0m0s
    Allows Msg Delete: true
         Allows Purge: true
       Allows Rollups: false

Limits:

     Maximum Messages: unlimited
  Maximum Per Subject: unlimited
        Maximum Bytes: unlimited
          Maximum Age: 1y0d0h0m0s
 Maximum Message Size: unlimited
    Maximum Consumers: unlimited

State:

             **Messages: 432**
                Bytes: 282 KiB
             FirstSeq: 1,133 @ 2024-06-20T12:48:59 UTC
              LastSeq: 359,493 @ 2024-06-26T15:31:41 UTC
     Deleted Messages: 357,929
     Active Consumers: 3
   Number of Subjects: 98`

And information about newly created consumer:

`Information for Consumer OrchestrationStream_08fa7d3dcd644745bace255d2c76fbc2 > orch_test created 2024-06-28T10:06:44+02:00

Configuration:

        Durable Name: orch_test
           Pull Mode: true
      Deliver Policy: All
          Ack Policy: All
            Ack Wait: 30s
       Replay Policy: Instant
     Max Ack Pending: 1,000
   Max Waiting Pulls: 512
        Headers Only: true

State:

   Last Delivered Message: Consumer sequence: 0 Stream sequence: 1,132
     Acknowledgment floor: Consumer sequence: 0 Stream sequence: 0
         Outstanding Acks: 0 out of maximum 1,000
     Redelivered Messages: 0
     **Unprocessed Messages: 563**
            Waiting Pulls: 0 of maximum 5`

How could I get actual information about consumer total messages count?
I can't get stream information, because consumers could have different subjects.

**Comment by scottf** [comment]
@Inkathu Is this reproducible? Can you please provide code to reproduce this, including your stream creation code and consumer creation code? My guess is that your consumer configuration subject filter covers more subjects than you expect.

**Reply by Inkathu** [reply]
This is the first time when we have had this kind of problem. So I don't know the steps to reproduce.
Even if this is misconfiguration - the stream has 432 messages - how could a consumer see more than the stream has?

**Reply by scottf** [reply]
I need some more info here. The data, consumer info, etc.

**Reply by Inkathu** [reply]
Here is the attached streams cache with one stream.
[nats_cache.zip](https://github.com/user-attachments/files/16032426/nats_cache.zip)
testorch - consumer created by nats cli, but you could create your own.
The problem that  stream contains : Messages: 5670
But the new consumer said:  5,803 / 100% unprocessed.
Is this info enough?
Server version: Version:  2.10.14

**Reply by Inkathu** [reply]
Created consumer settings:
![image](https://github.com/nats-io/nats.net/assets/15924269/52a43973-5130-46fb-9877-8e271afb8700)

**Reply by scottf** [reply]
Ok, this is not a client issue. I'm asking someone to help me here, I suspect that there is some explanation.

**Reply by scottf** [reply]
How did you populate the stream? How were the deletes done? Can you make a fresh stream, do all the adds and deletes and reproduce this? I'm just trying to get as much information so I can talk to the server team.

**Comment by scottf** [comment]
I noticed you published to a subject like this
```
OrchestrationStream_08fa7d3dcd644745bace255d2c76fbc2.>
```
You actual subject has a dot greater than sign at the end? What are you trying to accomplish. This may be the clue to the problem here. I would also recommend that you don't do that.

**Comment by scottf** [comment]
@Inkathu I'm closing this discussion. I have filed a bug in the server: https://github.com/nats-io/nats-server/issues/5608
